### PR TITLE
Fix cancel user invitation

### PR DESCRIPTION
**Pull request by AntoineDupre**
#1271

### CODE REVIEW

File: tests/core/gui/test_invite_user.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
```
Comment (touilleMan) [inline tests/core/gui/test_invite_user.py:1]: should move this test into `core/gui/users_widget`

File: tests/core/gui/test_invite_user.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+from parsec.core.gui.lang import translate as _
+
+from tests.common import customize_fixtures
+
+
+@pytest.mark.gui
+@pytest.mark.trio
+@customize_fixtures(logged_gui_as_admin=True)
+async def test_cancel_user_invitation(
+    qapp, aqtbot, logged_gui, running_backend, backend, autoclose_dialog, monkeypatch, alice
```
Comment (touilleMan) [inline tests/core/gui/test_invite_user.py:16]: `qapp` fixture needed ?

File: tests/core/gui/test_invite_user.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    # Patch dialogs
+    monkeypatch.setattr("parsec.core.gui.users_widget.get_text_input", lambda *x, **y: email)
+    monkeypatch.setattr(
+        "parsec.core.gui.users_widget.ask_question",
+        lambda *x, **y: _("TEXT_USER_INVITE_CANCEL_INVITE_ACCEPT"),
+    )
+    u_w = await logged_gui.test_switch_to_users_widget()
+
+    # Force send_email to false
+    _invite_new = u_w.core._backend_conn.cmds.invite_new
```
Comment (touilleMan) [inline tests/core/gui/test_invite_user.py:30]: no need for this you can use the decorator `@customize_fixtures(backend_has_email=True)`, then the fixture `email_letterbox` (see `tests/core/gui/users_widgets/test_invite.py`)

File: tests/core/gui/test_invite_user.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    async with aqtbot.wait_signals(
+        [
+            user_invitation_w.button_cancel.clicked,
+            user_invitation_w.cancel_clicked,
+            u_w.cancel_invitation_success,
+        ]
+    ):
+        await aqtbot.mouse_click(user_invitation_w.button_cancel, QtCore.Qt.LeftButton)
+
+    # Force gui update
```
Comment (touilleMan) [inline tests/core/gui/test_invite_user.py:61]: Do we need this ? I would say the list should be updated once we've clicked on the cancel button (hence no need to click again on the list users button)

File: parsec/core/gui/users_widget.py [hunk 5]
```diff
@@ -170,7 +170,7 @@ async def _do_list_users_and_invitations(core):
 
 async def _do_cancel_invitation(core, token):
     try:
-        await core.delete_invitation(token=token)
+        await core.delete_invitation(token=token, reason=InvitationDeletedReason.CANCELLED)
```
Comment (touilleMan) [inline parsec/core/gui/users_widget.py:173]: Funny thing is `core.delete_invitation` doesn't use the `reason` parameters ! https://github.com/Scille/parsec-cloud/blob/39866ad773c2333dc10b2cdb8a1ff8b7b553b099/parsec/core/logged_core.py#L206-L208

I think we should modify `core.delete_invitation` to use the `reason` param and to provide it with a default value of `InvitationDeletedReason.CANCELLED` (hence nothing to change in `_do_cancel_invitation`)

File: tests/core/gui/test_invite_user.py (hunk 3 above)
Comment (AntoineDupre) [reply tests/core/gui/test_invite_user.py:30]: it's much better :)

File: tests/core/gui/test_invite_user.py (hunk 4 above)
Comment (AntoineDupre) [reply tests/core/gui/test_invite_user.py:61]: yes it is better to wait on signal `u_w.list_success` to be sure that reset has been done

File: tests/core/gui/test_invite_user.py (hunk 2 above)
Comment (AntoineDupre) [reply tests/core/gui/test_invite_user.py:16]: removed

File: tests/core/gui/test_invite_user.py (hunk 1 above)
Comment (AntoineDupre) [reply tests/core/gui/test_invite_user.py:1]: i moved the test into `core/gui/users_widget/test_invite.py`

File: parsec/core/gui/users_widget.py (hunk 5 above)
Comment (AntoineDupre) [reply parsec/core/gui/users_widget.py]: fixed

**Pull request merged**